File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/Thermostat.py
class Thermostat:
    def __init__(self, current_temperature: float, target_temperature: float, mode: str):
        self.currentTemperature = current_temperature
        self.targetTemperature = target_temperature
        self.mode = mode

    def getTargetTemperature(self) -> float:
        return self.targetTemperature

    def setTargetTemperature(self, temperature: float) -> None:
        self.targetTemperature = temperature

    def getMode(self) -> str:
        return self.mode

    def setMode(self, mode: str) -> bool:
        if mode == "heat" or mode == "cool":
            self.mode = mode
            return True
        else:
            return False

    def autoSetMode(self) -> None:
        if self.currentTemperature < self.targetTemperature:
            self.mode = "heat"
        else:
            self.mode = "cool"

    def autoCheckConflict(self) -> bool:
        if self.currentTemperature > self.targetTemperature:
            if self.mode == "cool":
                return True
            else:
                self.autoSetMode()
                return False
        else:
            if self.mode == "heat":
                return True
            else:
                self.autoSetMode()
                return False

    def simulateOperation(self) -> int:
        self.autoSetMode()
        use_time = 0
        if self.mode == "heat":
            while self.currentTemperature < self.targetTemperature:
                self.currentTemperature += 1.0
                use_time += 1
        else:
            while self.currentTemperature > self.targetTemperature:
                self.currentTemperature -= 1.0
                use_time += 1
        return use_time

File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/java_to_cpp/Thermostat.cpp
#include <string>

class Thermostat {
public:
    double currentTemperature;
    double targetTemperature;
    std::string mode;

    Thermostat(double currentTemperature, double targetTemperature, std::string mode) {
        this->currentTemperature = currentTemperature;
        this->targetTemperature = targetTemperature;
        this->mode = mode;
    }

    double get_target_temperature() const {
        return targetTemperature;
    }

    void set_target_temperature(double temperature) {
        targetTemperature = temperature;
    }

    std::string get_mode() const {
        return mode;
    }

    bool set_mode(std::string mode) {
        if (mode == "heat" || mode == "cool") {
            this->mode = mode;
            return true;
        }
        return false;
    }

    void auto_set_mode() {
        if (currentTemperature < targetTemperature) {
            this->mode = "heat";
        } else {
            this->mode = "cool";
        }
    }

    bool auto_check_conflict() {
        if (currentTemperature > targetTemperature) {
            if (mode == "cool") {
                return true;
            } else {
                auto_set_mode();
                return false;
            }
        } else if (currentTemperature < targetTemperature) {
            if (mode == "heat") {
                return true;
            } else {
                auto_set_mode();
                return false;
            }
        }
        return false;
    }

    int simulate_operation() {
        auto_set_mode();
        int useTime = 0;
        if (mode == "heat") {
            while (currentTemperature < targetTemperature) {
                currentTemperature += 1.0;
                useTime += 1;
            }
        } else {
            while (currentTemperature > targetTemperature) {
                currentTemperature -= 1.0;
                useTime += 1;
            }
        }
        return useTime;
    }
};